lib/Catalyst/Plugin/Session/Store/MusicBrainz.pm

```diff
@@ -30,15 +30,11 @@ sub get_session_data {
 sub store_session_data {
     my ($self, $key, $data) = @_;
 
-    if (my ($sid) = $key =~ /^expires:(.*)/)
-    {
-        $self->_datastore->set($key, $data);
-    }
-    else
-    {
-        $self->_datastore->set($key, encode_base64(nfreeze($data)));
-        $self->_datastore->expireat($key, $self->session_expires);
+    unless ($key =~ /^expires:/) {
+        $data = encode_base64(nfreeze($data));
     }
+    $self->_datastore->set($key, $data);
+    $self->_datastore->expireat($key, $self->session_expires);
 }
```

Comment:
This is the main fix: Call `expireat` for *all* keys.
Plus some removal of duplicate code.